spec/controllers/queue_items_controller_spec.rb

```diff
@@ -0,0 +1,106 @@
+      get :index
+      expect(assigns(:queue_items)).to match_array([queue_item1, queue_item2])
+    end
+    
+    it 'redirects to the sign in page for unauthenticated users' do
+      get :index
+      expect(response).to redirect_to signin_path
+    end
+  end
+  
+  describe 'POST create' do 
+    let(:user1){ Fabricate(:user) }
+    let(:video){ Fabricate(:video) }
+    
+    it 'redirects to the my queue page' do 
+      session[:user_id] = user1.id 
+      post :create, video_id: Fabricate(:video).id, user_id: user1.id
+      expect(response).to redirect_to my_queue_path   
+    end
+    
+    it 'creates a queue item' do
+      session[:user_id] = user1.id 
+      post :create, video_id: Fabricate(:video).id, user_id: user1.id
+      expect(QueueItem.count).to eq(1)
+    end
+    
+    it 'creates the queue item that is associated with the video' do 
+      session[:user_id] = user1.id
+      post :create, video_id: video.id, user_id: user1.id
+      expect(assigns(:queue_item).video).to eq(video)
+    end
+    
+    it 'creates the queue item that is associated with the signed in user' do 
+      session[:user_id] = user1.id
+      post :create, video_id: video.id, user_id: user1.id
+      expect(assigns(:queue_item).user).to eq(user1)
+    end 
+    
+    it 'adds the queue item as last in the order' do 
+      session[:user_id] = user1.id
+      Fabricate(:queue_item, video_id: video.id, user_id: user1.id)
+      simpsons = Fabricate(:video)
+      post :create, video_id: simpsons.id, user_id: user1.id
+      simpsons_queue_item = QueueItem.where(video_id: simpsons.id, user_id: user1.id).first
+      expect(simpsons_queue_item.position).to eq(2)
+    end
+    
+    it 'does not add the video if the video is already in the queue' do 
+      my_item = Fabricate(:queue_item, video_id: video.id, user_id: user1.id)
+      session[:user_id] = user1.id
+      post :create, video_id: video.id, user_id: user1.id
+      expect(user1.queue_items.count).to eq(1)
+    end
+    
+    it 'redirects to the signin page for unauthenticated users' do 
+      post :create
+      expect(response).to redirect_to signin_path
+    end
+  end
+  
+  describe 'DELETE destroy' do
+    let(:user1){ Fabricate(:user) }
+    let(:video){ Fabricate(:video) }
+    let(:my_item){ Fabricate(:queue_item, video_id: video.id, user_id: user1.id) }
+
+    it 'removes the queue item from the database' do 
+      session[:user_id] = user1.id
+      delete :destroy, id: my_item.id 
+      expect(QueueItem.count).to eq(0) 
+    end 
+    
+    it 'redirects back to the myqueue page' do 
+      session[:user_id] = user1.id
+      delete :destroy, id: my_item.id 
+      expect(response).to redirect_to my_queue_path
+    end
+    
+    it 'does not let a user delete queue_items that are not in their queue' do 
+      user2 = Fabricate(:user)
+      user_1_item = Fabricate(:queue_item, video_id: video.id, user_id: user1.id)
```

Comment:
you should be able to do this `user1_item = Fabricate(:queue_item, video: video, user: user1)` - it cuts you some chars off :)